### DISCUSSION TITLE
How to change the visibility of a github pages site?

### DISCUSSION THREAD

**Discussion by dawhite1999** [community/community#54903] [Enterprise]
### Select Topic Area

Question

### Body

I know there is an [article](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site) for this on github already, but it doesn't quite answer my question. 

I have a private repository, in which I want to make a private github pages site. I need enterprise for this, so I upgraded to enterprise. However I'm still not able to change the visibility of the site. The github article mentions a dropdown menu with visibility settings under the github pages section of the settings, but I still just see the link to upgrade to enterprise.

I also tried to go to my enterprise page, and view the pages section there as well, but that requires me to add a domain. Is this a necessary step? or am I looking in the wrong place?

**Comment by yoannchaudet** [comment]
You should see the options under the Pages settings for your project sites only. That is, your repositories that are not named with the following pattern: `<org name>.github.io`. If that's not the case, [reach out to support](https://support.github.com/contact?tags=dotcom-footer) and we can take a look for you.

**Comment by kboucher** [comment]
I do not understand this accepted answer. If _GitHub Pages_ sites are published from repositories with the `<org name>.github.io` naming scheme, but we cannot access the _GitHub Pages visibility_ dropdown from a repository with that naming scheme, how are we supposed to make our GitHub Pages site private?

This alert box seems to suggest that upgrading to Enterprise Cloud should allow us to make our GitHub Pages site private:
<img width="730" alt="image" src="https://github.com/community/community/assets/1943568/6c8f1809-15d3-41fa-a89e-5e5d2cb4e57a">

https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site

**Reply by yoannchaudet** [reply]
Rename your repository to something other than `<org name>.github.io`. This repository has a special meaning for GitHub Pages, we call it your user/org site. Making this site private would potentially take down all your other sites (we call those project sites).

We talk briefly about the various type of sites in https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites.

**Reply by kboucher** [reply]
That was very helpful, thank you. The link you posted regarding the publishing of different types of GitHub Pages sites cleared up my confusion.

**Reply by oshriza** [reply]
Hi @yoannchaudet / @kboucher 

We have an Enterprise account (and I'm part of the owners of this Enterprise account) connected to my organization `(DevOps)`, Within this organization, I've created a private repository named `docs`, from which I've published a GitHub Pages site.

However, I've noticed that this GitHub Pages site is publicly accessible, despite the repository itself being private. I expected the GitHub Pages site to be accessible only to authenticated members of our enterprise.
The enterprise is connected with AzureAD, so all the members are connecting to their account like: `oshriz_{org_name}`

Could you please guide me on how to restrict access to this GitHub Pages site so that it is only accessible to members of our enterprise?

Thanks in advance for your help!
